src/lib/concurrency/transaction_manager.hpp

```diff
@@ -56,7 +56,7 @@ class TransactionManager : public Noncopyable {
   /**
    * Creates a new transaction context
    */
-  std::shared_ptr<TransactionContext> new_transaction_context();
+  std::shared_ptr<TransactionContext> new_transaction_context(bool is_auto_commit = true);
```

Comment:
Does it make sense for this to be the default?

Reply:
One argument could be: Auto-commit Transactions are implicitly created without a specific user command. Now these transactions don't explicitly need to state their implicity during creation, since auto-commit is the standard behavior. 

Additionally Auto-Commit is far more common in the codebase.

Reply:
1) Does the argument convince you?
2) Or should we remove the default for readability?

I prefer the second.

Reply:
Agreed. I don't know of a general "default argument rule". My personal approach is to use defaults only when the default is painfully obvious or when the usual caller would simply not care about it. Here, neither applies.